xl_auth/public/views.py

```diff
@@ -43,10 +43,54 @@ def home():
             return redirect(redirect_url)
         else:
             flash_errors(login_form)
+
     return render_template('public/home.html', login_form=login_form,
                            next_redirect_url=request.args.get('next'))
 
 
+@blueprint.route('/forgot_password/', methods=['GET', 'POST'])
+def forgot_password():
+    """Forgot password form."""
+    forgot_password_form = ForgotPasswordForm(request.form)
+    if request.method == 'POST':
+        if forgot_password_form.validate_on_submit():
+            user = User.get_by_email(forgot_password_form.username.data)
+            password_reset = PasswordReset(user)
+            password_reset.save()
+            # TODO: Do stuff to send out email with code.
+            flash(_('This is your reset code %(code)s.', code=password_reset.code), 'success')
+            # FIXME: Must be removed before deploying to prod.
+            flash('Reset link: %s' % url_for('public.reset_password', email=user.email,
+                                             code=password_reset.code, _external=True), 'warning')
+            return redirect(url_for('public.home'))
+        else:
+            flash_errors(forgot_password_form)
+
+    return render_template(
+        'public/forgot_password.html', forgot_password_form=forgot_password_form)
+
+
+@blueprint.route('/reset_password/<string:email>/<string:code>', methods=['GET', 'POST'])
```

Comment:
Why use args as part of the path instead of as query params? I have a feeling query params are more common, so I'm interested in the reasoning.

Reply:
@mxtthias I have the same feeling.. Maybe it's a matter of framework?

My reasoning was it made sense because both are strictly mandatory. There's no "reset password" without a) target username and b) a reset secret.

I think query params are mostly used for things that you can opt-in for. And it also makes it straight-forward using the built-in URL handling, e.g.
```python
url_for('public.reset_password', email=password_reset.user.email, code=password_reset.code)
```

Instead of string-formatting and encoding it manually. :)